$OpenBSD: patch-source3_wscript,v 1.13 2021/10/18 12:25:20 jca Exp $

- avoid using glib (only useful for spotlight?)
- use more generic check for OpenBSD.
- avoid nested function error with clang

Index: source3/wscript
--- source3/wscript.orig
+++ source3/wscript
@@ -100,6 +100,7 @@ def options(opt):
 
     # default = None means autodetection
     opt.samba_add_onoff_option('spotlight', with_name="enable", without_name="disable", default=None)
+    opt.samba_add_onoff_option('glib', default=False)
 
 def configure(conf):
     default_static_modules = []
@@ -112,7 +113,7 @@ def configure(conf):
         conf.ADD_CFLAGS('-DDEVELOPER -DDEBUG_PASSWORD')
         conf.env.developer = True
 
-    if sys.platform != 'openbsd5':
+    if not sys.platform.startswith('openbsd'):
         conf.ADD_LDFLAGS("-Wl,--export-dynamic", testflags=True)
 
     # We crash without vfs_default
@@ -1677,6 +1678,7 @@ main() {
 
     conf.CHECK_CODE('void seekdir(DIR *d, long loc) { return; }',
                     'SEEKDIR_RETURNS_VOID',
+                    addmain=False,
                     headers='sys/types.h dirent.h',
                     msg='Checking whether seekdir returns void')
 
@@ -1790,7 +1792,7 @@ main() {
         else:
             conf.fatal('AFS headers not available, but --with-fake-kaserver was specified')
 
-    if conf.CHECK_CFG(package='glib-2.0',
+    if Options.options.with_glib and conf.CHECK_CFG(package='glib-2.0',
                       args='--cflags --libs',
                       msg='Checking for glib-2.0',
                       uselib_store="GLIB-2.0"):
